### ISSUE TITLE
Format `location` properties as numbers

### ISSUE THREAD

**Issue by Kageetai** [jose-elias-alvarez/obsidian-geocoding-properties#4]
Hey, I was just comparing this plugin to the [obsidian map view plugin](https://github.com/esm7/obsidian-map-view), which has similar functionality, but I like how your plugin adds more info to the notes.
But I did notice one major difference when adding the location info to the notes. Obsidian Map View formats the location property as numbers, but your plugin does so as strings.
I understand why you probably decided to use strings, as it shows the property properly within the Obsidian property tool and still keeps it editable, as this is exactly [a problem Obsidian Map View actually has](https://github.com/esm7/obsidian-map-view/issues/202).
But this difference makes it incompatible to use your plugin with Obsidian Map View (although the Leaflet plugin still works), so I was wondering whether you could add a setting for the user to choose which format to apply when adding the frontmatter. Like let the user choose whether to format the values as strings or not.
Would that be possible? Thx

https://github.com/jose-elias-alvarez/obsidian-geocoding-properties/blob/c501cb31dae02d99651308a2b8b5bcda4b46c93a/src/main.ts#L97-L98

**Comment by jose-elias-alvarez** [comment]
Hmm, I'm not necessarily opposed to adding a new setting or a new property to handle this, but as you noted, using numbers instead of strings makes the property uneditable in Obsidian, since it doesn't know how to handle it:

![image](https://github.com/jose-elias-alvarez/obsidian-geocoding-properties/assets/54108223/38429ff0-0126-436e-8956-09311e93b6eb)

Are you sure it wouldn't be better to solve this issue in Obsidian Map View?

**Comment by Kageetai** [comment]
Yes, I know that neither is ideal, I just thought that the format with numbers instead of strings also somewhat makes more sense, as coordinates are numbers after all.
In the end, it would be good to be able to create custom property formats in Obsidian itself, as someone in the thread I linked above also [suggested](https://github.com/esm7/obsidian-map-view/issues/202#issuecomment-1719457180). But in the meantime, either plugin developers need to come up with their own fixes :/
Anyway, not a hard problem for me, just thought this side would be the easiest fix

**Comment by jose-elias-alvarez** [comment]
If for whatever reason the issue can't be solved in obsidian-map-view (the maintainer is unwilling or the plugin is unmaintained) then I'm open to adding an option here. (I see that the repo owner tagged the issue you linked in January, so hopefully putting in a fix there is an option, since I suspect it'll take quite a while before Obsidian allows custom property types.)

**Comment by Kageetai** [comment]
Yeah, makes sense. Maybe they can fix it there

**Comment by evanharmon1** [comment]
The map view plugin has a new version in the works related to these issues - https://github.com/esm7/obsidian-map-view/tree/v5.0

My opinion is that data like [lat,long] should be an array since it's easier to parse that way and I think Obsidian properties get confused in some cases if the literal `[lat,long]` string is in a string type property since it thinks its an array. So that would be my preference for how a plugin saves location data. But good config options for how the plugin does it would be great so we could do whatever.

Thanks for the plugin!

**Comment by sandervdvijver** [comment]
An option in the config would be ideal for me to, as I would like to store them as `[lat,long]`as well.
